ZelBack/src/services/appsService.js

```diff
@@ -14254,101 +14378,158 @@ async function checkMyAppsAvailability() {
       pubKey,
     };
     const stringData = JSON.stringify(data);
-    // eslint-disable-next-line no-await-in-loop
     const signature = await signCheckAppData(stringData);
     data.signature = signature;
-    // first check against our IP address
-    // eslint-disable-next-line no-await-in-loop
-    const resMyAppAvailability = await axios.post(`http://${askingIP}:${askingIpPort}/flux/checkappavailability`, JSON.stringify(data), axiosConfig).catch(async (error) => {
-      log.error(`checkMyAppsAvailability - ${askingIP} for app availability is not reachable`);
-      log.error(error);
-      setPortToTest = testingPort;
-      failedNodesTestPortsCache.set(askingIP, askingIP);
-      await serviceHelper.delay(30 * 1000);
-      return checkMyAppsAvailability();
-    });
-    if (resMyAppAvailability && resMyAppAvailability.data.status === 'error') {
-      log.warn(`checkMyAppsAvailability - Applications port range unavailability detected from ${askingIP}:${askingIpPort} on ${testingPort}`);
+
+    // this will be problematic in the future (when we fix the api to respect
+    // content-type headers). As we are stringifying the data, axios doesn't
+    // add the appropriate header. (i.e. stop stringifying)
+    const resMyAppAvailability = await axios
+      .post(
+        `http://${askingIP}:${askingIpPort}/flux/checkappavailability`,
+        JSON.stringify(data),
+        axiosConfig,
+      )
+      .catch(() => {
+        log.error(
+          `checkMyAppsAvailability - ${askingIP} for app availability is not reachable`,
+        );
+        setPortToTest = testingPort;
+        failedNodesTestPortsCache.set(askingIP, askingIP);
+        return null;
```

Comment:
if we return null this will stop be executing, we need to call the function again.